src/main/scala/mesosphere/marathon/core/election/impl/CuratorElectionService.scala

```diff
@@ -125,6 +126,25 @@ class CuratorElectionService(
           CurrentRuntime.asyncExit()(scala.concurrent.ExecutionContext.global)
           false
         }
+      }).zookeeperFactory(new ZookeeperFactory {
+        val delegate = new DefaultZookeeperFactory
+        override def newZooKeeper(
+          connectString: String,
+          sessionTimeout: Int,
+          watcher: Watcher,
+          canBeReadOnly: Boolean): ZooKeeper =
+
+          try {
+            delegate.newZooKeeper(connectString, sessionTimeout, watcher, canBeReadOnly)
+          } catch {
+            // We may see exceptions here, for example, when none of the ZK hosts are resolvable at Marathon startup;
+            // in this case Curator may fail with some background exception and Marathon never offers leadership.
+            case NonFatal(e) =>
+              // TODO(jdef) should we really be committing suicide here? what about retries instead?
+              log.error("ZooKeeper initialization failed - Committing suicide")
+              CurrentRuntime.asyncExit()(scala.concurrent.ExecutionContext.global)
+              null
```

Comment:
We hard assert the zk urls are resolvable / spartan is up before we start Marathon on DC/OS:
https://github.com/dcos/dcos/blob/master/packages/marathon/build#L26

So marathon starting but the magic url not being resolvable shouldn't be possible.